src/Searchbar.vue

```diff
@@ -35,18 +40,57 @@ export default {
             });
           }
         });
-        if (!hasMatchingHeading) {
-          if (regex.test(title) || regex.test(keywords)) {
-            matches.push(entry);
-          }
-        }
       });
       return matches;
     },
   },
   filters: {
     highlight(value, phrase) {
-      return value.replace(new RegExp(`(${phrase})`, 'gi'), '<mark>$1</mark>');
+      function mergeOverlappingIntervals(intervals) {
+        if (intervals.length <= 1) {
+          return intervals;
+        }
+
+        const sortedIntervals = intervals.sort((a, b) => a.start - b.start);
+        const stack = [];
+        stack.push(sortedIntervals[0]);
+        for (let i = 1; i < sortedIntervals.length; i += 1) {
+          const top = stack[stack.length - 1];
+
+          if (top.end < sortedIntervals[i].start) {
+            stack.push(sortedIntervals[i]);
+          } else if (top.end < sortedIntervals[i].end) {
+            top.end = sortedIntervals[i].end;
+          }
+        }
+
+        return stack;
+      }
+
+      function collectMatchPos() {
+        const keywords = phrase.split(' ').filter(keyword => keyword !== '');
+        const matchPos = [];
+        let match;
+        keywords.forEach((keyword) => {
+          const regex = new RegExp(`(${keyword})`, 'gi');
+          match = regex.exec(value);
+          while (match != null) {
+            matchPos.push({ start: match.index, end: match.index + keyword.length });
+            match = regex.exec(value);
+          }
+        });
+        return matchPos;
+      }
+
+      const matchPos = mergeOverlappingIntervals(collectMatchPos());
+
+      let result = value;
+      for (let i = matchPos.length - 1; i >= 0; i -= 1) {
```

Comment:
Add a comment to explain why we are processing it in reverse (to avoid the positioning going out of sync if we modify the front first vs. the back).

---

Alternatively, I just thought of another algorithm that we could possibly use: split the string up according to the intervals, add the correct mark tags before joining them back.

E.g.
Suppose the title is "This is some text", and the search keywords are "This and is".
Once you found the intervals, split them up into an array according to the intervals.

`['This', ' ', 'is', ' some text'];`

Then insert the necessary marks and then rejoin them (the insertion can be done during the splitting as well rather than after the splitting if it makes it easier).

`['<mark>', 'This', '</mark>', ' ', '<mark>', 'is', '</mark>', ' some text'].join('')`

What do you think about this approach?